dashboard/app/controllers/projects_controller.rb

```diff
@@ -328,7 +328,9 @@ def get_or_create_for_level
 
     if user_id
       # If viewing another user's work, ensure that we have permission.
-      script_level = level.script_levels.find_by_script_id(script_id)
+      script_level = params[:script_level_id] ?
+        ScriptLevel.cache_find(params[:script_level_id].to_i) :
+        level.script_levels.find_by_script_id(script_id)
```

Comment:
Is there a potential malicious case in which the request provides a `script_level_id` which  is used here to check for permission, but which is unrelated to the `level_id` which is used below to actually retrieve the channel?

Reply:
Oh true, that's a good point...yeah it would make sense to have a stricter check here. Ideally we should be able to use the script ID in combination with sublevel ID to find the script level ID. I'll look into that.

Reply:
Updated this with a check in `projects_controller` which ensures that if a script level ID is provided, it either matches the provided level ID or is associated with a valid parent level of provided sublevel ID. Added a unit test for this too. Let me know if you think this feels strict enough!

Reply:
I don't know these subsystems very well, but that seems like a good improvement, with good test coverage too, thanks!